idl/github.com/uber/cadence/shared.thrift

```diff
@@ -835,3 +826,10 @@ struct WorkflowQuery {
   10: optional string queryType
   20: optional binary queryArgs
 }
+
+struct RespondQueryTaskCompletedRequest {
+  10: optional binary taskToken
+  20: optional TaskList taskList
```

Comment:
Probably this can be encoded in the tasktoken.

Reply:
good call